h5py/h5t.pyx

```diff
@@ -247,6 +247,13 @@ H5Tlock(H5PY_OBJ)
 
 PYTHON_OBJECT = lockid(H5PY_OBJ)
 
+# Datetime64
+cdef hid_t H5PY_DATETIME64 = H5Tcreate(H5T_OPAQUE, 8)
+H5Tset_tag(H5PY_DATETIME64, "NUMPY:DATETIME64")
+H5Tlock(H5PY_DATETIME64)
+
+NUMPY_DATETIME64 = lockid(H5PY_DATETIME64)
```

Comment:
I think we need more information about precision and endinaness encoded, which might mean more HT5s.  For example,  `np.datetime64('2017-11-29T17:11:44.638713')` has `dtype('<M8[us]')` and this could / would be read in incorrectly if you built a file with one native endianness and brought it to a system with a different endianness.